Framework/PythonInterface/mantid/plots/mantidaxes.py

```diff
@@ -418,17 +439,39 @@ def replot_artist(self, artist, errorbars=False, **kwargs):
         """
         kwargs['distribution'] = not self.get_artist_normalization_state(artist)
         workspace, spec_num = self.get_artists_workspace_and_spec_num(artist)
-        self.remove_artists_if(lambda art: art == artist)
-        if kwargs.get('axis', None) == MantidAxType.BIN:
-            workspace_index = spec_num
+
+        #check if it is a sample log plot
+        if spec_num is None:
+            #error bars never make sense for sample log plots
+            sample_log_plot_details = self.get_artists_sample_log_plot_details(artist)
+            kwargs['LogName'] = sample_log_plot_details[0]
+            if sample_log_plot_details[1] is not None:
+                kwargs['Filtered'] = sample_log_plot_details[1]
+            if sample_log_plot_details[2] is not None:
+                kwargs['ExperimentInfo'] = sample_log_plot_details[2]
+            # error bar plots do not make sense for log plots
+            errorbars=False
+            # neither does distribution
+            if 'distribution' in kwargs.keys():
+                    del kwargs['distribution']
         else:
-            workspace_index = workspace.getIndexFromSpectrumNumber(spec_num)
-        self._remove_matching_curve_from_creation_args(workspace.name(), workspace_index, spec_num)
+            if kwargs.get('axis', None) == MantidAxType.BIN:
+                workspace_index = spec_num
+            else:
+                workspace_index = workspace.getIndexFromSpectrumNumber(spec_num)
 
+            self._remove_matching_curve_from_creation_args(workspace.name(), workspace_index, spec_num)
+            kwargs['wkspIndex'] = workspace_index
+
+        self.remove_artists_if(lambda art: art == artist)
         if errorbars:
-            new_artist = self.errorbar(workspace, wkspIndex=workspace_index, **kwargs)
+            new_artist = self.errorbar(workspace, **kwargs)
         else:
-            new_artist = self.plot(workspace, wkspIndex=workspace_index, **kwargs)
+            # if it is a sample log plot
+            if spec_num is None:
+                new_artist = self.plot(workspace, **kwargs)[0]
+            else:
+                new_artist = self.plot(workspace, **kwargs)[0]
```

Comment:
You have the if and the else both doing the same thing
```suggestion
            new_artist = self.plot(workspace, **kwargs)[0]
```